docs/shared/concepts/typescript-project-linking.md

```diff
@@ -0,0 +1,308 @@
+{
+  "devDependencies": {
+    "@my-org/some-project": "*"
+  }
+}
+```
+
+{% /tab %}
+{% tab label="bun" %}
+
+```json {% fileName="package.json" %}
+{
+  "workspaces": ["apps/**", "packages/**"]
+}
+```
+
+Defining the `workspaces` property in the root `package.json` file lets bun know to look for other `package.json` files in the specified folders. With this configuration in place, all the dependencies for the individual projects will be installed in the root `node_modules` folder when `bun install` is run in the root folder. Also, the projects themselves will be linked in the root `node_modules` folder to be accessed as if they were npm packages.
+
+If you want to reference a local library project with its own `build` task, you should include the library in the `devDependencies` of the application's `package.json` with `*` specified as the library's version. `*` tells bun to use whatever version of the project is available.
+
+```json {% fileName="/apps/my-app/package.json" %}
+{
+  "devDependencies": {
+    "@my-org/some-project": "*"
+  }
+}
+```
+
+{% /tab %}
+{% tab label="pnpm" %}
+
+```yaml {% fileName="pnpm-workspace.yaml" %}
+packages:
+  # specify a package in a direct subdir of the root
+  - 'my-app'
+  # all packages in direct subdirs of packages/
+  - 'packages/*'
+  # all packages in subdirs of components/
+  - 'components/**'
+  # exclude packages that are inside test directories
+  - '!**/test/**'
+```
+
+Defining the `packages` property in the root `pnpm-workspaces.yaml` file lets pnpm know to look for project `package.json` files in the specified folders. With this configuration in place, all the dependencies for the individual projects will be installed in the root `node_modules` folder when `pnpm install` is run in the root folder.
+
+If you want to reference a local library project from an application, you need to include the library in the `devDependencies` of the application's `package.json` with `workspace:*` specified as the library's version. `workspace:*` tells pnpm that the project is in the same repository and not an npm package. You want to specify local projects as `devDependencies` instead of `dependencies` so that the library is not included twice in the production bundle of the application.
+
+```json {% fileName="/apps/my-app/package.json" %}
+{
+  "devDependencies": {
+    "@my-org/some-project": "workspace:*"
+  }
+}
+```
+
+{% /tab %}
+{% /tabs %}
+
+### Set Up TypeScript Project References
+
+With workspaces enabled, you can also configure TypeScript project references to speed up your build and typecheck tasks.
+
+The root `tsconfig.base.json` should contain a `compilerOptions` property and no other properties. `compilerOptions.composite` and `compilerOptions.declaration` should be set to `true`. `compilerOptions.paths` should not be set.
+
+```jsonc {% fileName="tsconfig.base.json" %}
+{
+  "compilerOptions": {
+    // Required compiler options
+    "composite": true,
+    "declaration": true,
```

Comment:
Setting `composite: true` enables `declaration: true` by default. It's not wrong to set it and it's required to have `declaration` enabled. So, the wrong setup is if the user already had `declaration: false`, in which case, they need to set it to `true` or remove the option so the default applies.

FWIW, we generate the `tsconfig.base.json` without setting `declaration`.